File: 1013-fibonacci-number/solution.cpp
class Solution {
public:
    int dp[50];
    int fib(int n) {
        
        if(n == 1 || n == 2) return 1 ;
        
        if(n < 1)return 0 ;
        
        if(dp[n] != 0) return dp[n];
        
        int x = fib( n - 1) ;
        int y = fib( n - 2) ;
        
        return dp[n] = x + y ;
        
    }
};

File: 1013-fibonacci-number/solution.java
class Solution {
    int[] dp = new int[50];
    public int fib(int n) {
        if(n < 2) return n ;

        if(dp[n] != 0) return dp[n];
        int x = fib(n-1);
        int y = fib(n-2);
        return dp[n] = x + y ;
    }
}
